File: models/track.go
package models

import (
	"time"

	"github.com/zmb3/spotify"
)

type Track struct {
	ID            string          `json:"id"`
	Name          string          `json:"name"`
	Artists       []TrackArtist   `json:"artists"`
	Images        []spotify.Image `json:"images"`
	BeganPlaying  time.Time       `json:"began_playing"`
	Duration      uint            `json:"duration"`
	Contributor   string          `json:"contributor"`
	ContributorID int             `json:"contributor_id"`
}

type TrackArtist struct {
	ID   string `json:"id"`
	Name string `json:"name"`
}

func NewTrack(t spotify.FullTrack) Track {
	track := Track{
		ID:            t.ID.String(),
		Name:          t.Name,
		Images:        t.Album.Images,
		Artists:       make([]TrackArtist, len(t.Artists)),
		BeganPlaying:  time.Unix(0, 0).UTC(),
		Duration:      uint(t.Duration / 1000),
		ContributorID: -1,
	}
	for i, artist := range t.Artists {
		track.Artists[i] = TrackArtist{
			ID:   artist.ID.String(),
			Name: artist.Name,
		}
	}
	return track
}

func NewTracks(t []spotify.PlaylistTrack) []Track {
	tracks := make([]Track, len(t))
	for i, track := range t {
		tracks[i] = NewTrack(track.Track)
	}
	return tracks
}

func (list *TrackList) Tracks() []Track {
	var tracks []Track
	list.Data.Unmarshal(&tracks)
	return tracks
}

File: Server/Models/Track.cs
using System;
using System.Collections.Generic;
using System.Linq;
using Newtonsoft.Json;
using Spotify.API.NetCore.Models;

namespace Server.Models
{
  public class Track
  {
    public Track()
    {
      Artists = new List<TrackArtist>();
      Images = new List<Image>();
    }

    [JsonProperty("id")]
    public string Id { get; set; }

    [JsonProperty("name")]
    public string Name { get; set; }

    [JsonProperty("artists")]
    public List<TrackArtist> Artists { get; set; }

    [JsonIgnore]
    public string FirstArtist => Artists
      .Select(a => a.Name)
      .DefaultIfEmpty("Unknown")
      .FirstOrDefault();

    [JsonProperty("images")]
    public List<Image> Images { get; set; }

    [JsonProperty("endpoint")]
    public string Endpoint { get; set; }

    [JsonProperty("began_playing")]
    public DateTime? BeganPlaying { get; set; }

    [JsonProperty("duration")]
    public int Duration { get; set; }

    [JsonProperty("contributor")]
    public string Contributor { get; set; }

    [JsonProperty("contributor_id")]
    public int ContributorID { get; set; }

    public static Track FromSpotify(FullTrack track) => new Track()
    {
      Id = track.Id,
      Name = track.Name,
      Artists = track.Artists.Select(a => new TrackArtist() { Id = a.Id, Name = a.Name}).ToList(),
      Images = track.Album.Images,
      Endpoint = track.Href,
      BeganPlaying = null,
      Duration = (int) Math.Round(track.DurationMs / 1000.0),
      Contributor = null,
      ContributorID = -1
    };
  }
}
